### PR TITLE
feat: auth client keyType config option

### PR DESCRIPTION
**Pull request by krpeacock**
# Description

Rem.codes pointed out that the ECDSA keys weren't working in a localstorage implementation. This feature adds the option to choose between ECDSA and ED25519 keys for the auth client base key.

It will also warn developers if they are using a custom storage implementation with "ECDSA" that it may not work correctly, and that they may want to consider using ED25519.

# How Has This Been Tested?

new unit test

# Checklist:

- [x] My changes follow the guidelines in [CONTRIBUTING.md](https://github.com/dfinity/agent-js/blob/main/CONTRIBUTING.md).
- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [x] I have edited the CHANGELOG accordingly.
- [x] I have made corresponding changes to the documentation.

### CODE REVIEW

File: packages/auth-client/src/index.ts [hunk 1]
```diff
@@ -203,10 +220,13 @@ export class AuthClient {
           const fallbackLocalStorage = new LocalStorage();
           const localChain = await fallbackLocalStorage.get(KEY_STORAGE_DELEGATION);
           const localKey = await fallbackLocalStorage.get(KEY_STORAGE_KEY);
-          if (localChain && localKey) {
+          // not relevant for Ed25519
+          if (localChain && localKey && keyType === 'ECDSA') {
             console.log('Discovered an identity stored in localstorage. Migrating to IndexedDB');
             await storage.set(KEY_STORAGE_DELEGATION, localChain);
+            localKey; //?
```
Comment (dfx-json) [inline packages/auth-client/src/index.ts:227]: what does this line do?

File: packages/auth-client/src/index.ts [hunk 2]
```diff
@@ -219,7 +239,11 @@ export class AuthClient {
       if (maybeIdentityStorage) {
         try {
           if (typeof maybeIdentityStorage === 'object') {
-            key = await ECDSAKeyIdentity.fromKeyPair(maybeIdentityStorage);
+            if (keyType === 'Ed25519' && typeof maybeIdentityStorage === 'string') {
```
Comment (dfx-json) [inline packages/auth-client/src/index.ts:242]: I wonder if strings like `Ed25519` and `ECDSA` would be better off defined as enums or consts.

**Review by dfx-json — APPROVED** [review]
Left a couple comments. but otherwise, LGTM

File: packages/auth-client/src/index.ts (hunk 1 above)
Comment (krpeacock) [reply packages/auth-client/src/index.ts:227]: oh, that's a comment. I'll remove it

File: packages/auth-client/src/index.ts (hunk 2 above)
Comment (krpeacock) [reply packages/auth-client/src/index.ts:242]: I can move it to a const

Comment (krpeacock) [reply packages/auth-client/src/index.ts:242]: although, that doesn't play well with the actual parameter

**Pull request merged**